frontend/js/src/user/playlists/components/ImportPlaylistModal.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useContext, useState } from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+import { ToastMsg } from "../../../notifications/Notifications";
+
+export const MUSICBRAINZ_JSPF_PLAYLIST_EXTENSION =
+  "https://musicbrainz.org/doc/jspf#playlist";
+
+type ImportPLaylistModalProps = {
+  listenlist: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPLaylistModalProps) => {
+  const modal = useModal();
+
+  const closeModal = React.useCallback(() => {
+    modal.hide();
+    document?.body?.classList?.remove("modal-open");
+    setTimeout(modal.remove, 200);
+  }, [modal]);
+
+  const { APIService, currentUser } = React.useContext(GlobalAppContext);
+  const [playlists, setPlaylists] = React.useState<SpotifyPlaylistObject[]>([]);
+
+  React.useEffect(() => {
+    async function importPlaylistFromSpotify() {
+      try {
+        const response = await APIService.importPlaylistToSpotify(
```

Comment:
Naming here can be improved. You're not importing the playlists per se.
Maybe `getUserPlaylistsFromSpotify` or something like that ?